common/lib/dependabot/experiments.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Dependabot
+  module Experiments
+    def self.register(name, value)
+      @experiments ||= {}
+      @experiments[name.to_sym] = value
+    end
```

Comment:
I really like this move to globalize the Experiment! Feels much safer than passing around a hash that can be mutated accidentally 😅

This interface reminds me somewhat of Flipper. In that case, `#register` is [scoped to groups](https://github.com/jnunemaker/flipper/blob/2dc8f6da8d75bc889cf262e7197b2d26a07a4f6d/lib/flipper.rb#L71-L91), which isn't really what we're after here. Of course, we don't have to follow that interface, but it might be worth considering. I could see `.toggle(name, should_enable)`, `.enable(name)` and `.disable(name)` or something along those lines.

I know this is a PR to be merged into a feature branch, but I thought it was worth mentioning 😇

Reply:
Yeah I'm definitely open to adding those methods, but I was thinking let's add them when we actually need the functionality?